benchadapt/python/benchadapt/adapters/gbench.py

```diff
@@ -164,20 +164,21 @@ def _parse_results(self, results: dict, extra_tags: dict) -> List[BenchmarkResul
         gbench_context, benchmark_groups = self._parse_gbench_json(results)
 
         parsed_results = []
-        for benchmark in benchmark_groups:
-            # all results for a benchmark name share a batch id
-            batch_id = uuid.uuid4().hex
-            result_parsed = self._parse_benchmark(
-                result=benchmark,
-                batch_id=batch_id,
-                extra_tags=extra_tags,
-            )
-            result_parsed.optional_benchmark_info = {"gbench_context": gbench_context}
-            parsed_results.append(result_parsed)
+        for name, benchmark in benchmark_groups.items():
```

Comment:
I'm not sure if this is better or worse, but `name` here goes unused, right? Would it be preferable to do `for _, benchmark in ...`? to show that?

Reply:
`name` does get used as a key on L219 (though really that could be a list instead of a dict; the keys are just so it's easier to see what each element is), but `name_params` on L223 never gets used. I left it so it's obvious what the grouping level here is (something like `MyBenchmark/param1=1/param2=2`), though by convention it should be `_`. I mildly prefer writing it out so it's easier to see why it's structured like this, but it's not a strong preference if someone disagrees.

Reply:
_nods_ yeah, I saw that it was used elsewhere, but was (mildly) confused about this specific loop here. Just to be extra clear: the loop down below that you mention (~L219) is totally separate from this one, yeah?

> I left it so it's obvious what the grouping level here is (something like MyBenchmark/param1=1/param2=2), though by convention it should be _. I mildly prefer writing it out so it's easier to see why it's structured like this, but it's not a strong preference if someone disagrees.

Maybe let's put a comment that says that about this loop? Just so that anyone coming back here doesn't think "hey this `name` in this loop definition is totally superfluous, that's dumb!"

Reply:
Renamed a bit and rearranged the comments to try to make it clearer what both the outer and inner grouping and loops are doing